### PR TITLE
device-tree: Fix issue for custom DTS

### PR DESCRIPTION
**Pull request by werneazc**
If someone uses the recommendations from the README.md file to create a custom device tree, the build runs into an error, because cp tries to use the same values for source and destination if "KERNEL_DTB_PATH" is set to "workdir". This results in a return value unequal to zero and stops executing the build process.

Moreover, the string comparison from USE_KERNEL_SOURCES did not contain double quotes, thus it was not interpreted as a string.

Signed-off-by: Andre Werner <andre.werner@systec-electronic.com>

### CODE REVIEW

File: meta-adi-xilinx/recipes-bsp/device-tree/device-tree.bbappend [hunk 1]
```diff
@@ -124,8 +124,11 @@ do_configure_append() {
 	# logic on the $dtb_tag_file without directly changing the original file (being it an in kernel
 	# devicetree or not). In case it's an in kernel devicetree, changing it would actually
 	# break the kernel compilation
-	[ ${USE_KERNEL_SOURCES} == "y" ] && cp -rf "${DTS_INCLUDE_PATH}/"* "${WORKDIR}/"
-	[ ${KERNEL_DTB_PATH} != ${DTS_INCLUDE_PATH} ] && cp -rf "${KERNEL_DTB_PATH}/"* "${WORKDIR}/"
+	[ "${USE_KERNEL_SOURCES}" = "y" ] && cp -Rf "${DTS_INCLUDE_PATH}/"* "${WORKDIR}/"
+
+	[ "${KERNEL_DTB_PATH}" != "${WORKDIR}" ] && \
+		[ "${KERNEL_DTB_PATH}" != "${DTS_INCLUDE_PATH}" ] && cp -Rf "${KERNEL_DTB_PATH}/"* "${WORKDIR}/"
```
Comment (nunojsa) [inline meta-adi-xilinx/recipes-bsp/device-tree/device-tree.bbappend:130]: Arghh, I'm fairly sure I already had caught this issue and somehow forgot to address it :facepalm: 

For now let's have this as-is. In the future, I'm not sure how useful it is to support custom devicetrees in external paths (i.e not directly in the kernel sources or not automatically copied to WORKDIR via SRC_URI). In that case we would not need the second `cp` at all...

File: meta-adi-xilinx/recipes-bsp/device-tree/device-tree.bbappend [hunk 2]
```diff
@@ -124,8 +124,11 @@ do_configure_append() {
 	# logic on the $dtb_tag_file without directly changing the original file (being it an in kernel
 	# devicetree or not). In case it's an in kernel devicetree, changing it would actually
 	# break the kernel compilation
-	[ ${USE_KERNEL_SOURCES} == "y" ] && cp -rf "${DTS_INCLUDE_PATH}/"* "${WORKDIR}/"
-	[ ${KERNEL_DTB_PATH} != ${DTS_INCLUDE_PATH} ] && cp -rf "${KERNEL_DTB_PATH}/"* "${WORKDIR}/"
+	[ "${USE_KERNEL_SOURCES}" = "y" ] && cp -Rf "${DTS_INCLUDE_PATH}/"* "${WORKDIR}/"
```
Comment (nunojsa) [inline meta-adi-xilinx/recipes-bsp/device-tree/device-tree.bbappend:127]: Hmmm there's a subtle change in here. I'm ok to keep the addition of the quotes in the same commit as this one but changing from `==` to `=` has more into it... Please check the discussion in this issue #102 

That said, the change to `=` is definitely valid but please do it in a separate commit with a proper explanation for it. Better not to hide it in this patch.

**Review by nunojsa — COMMENTED** [review]
Hi,

Thanks a lot for the fixes... Just some minor notes from me. 

Also I have the same procedure as in linux kernel for fixes so it would be nice if you could add a Fixes: tag in the commit description (just git blame the file to find the culprit commit; should be the initial one). Look [here](https://github.com/analogdevicesinc/meta-adi/commit/15481781f99bcef61ba0e58bf4bdb25e8cf1df41) for an example...

**Review by werneazc — COMMENTED** [review]
Reviewed the pull request regarding the suggestions.
